lib/ansible/plugins/strategy/__init__.py

```diff
@@ -385,6 +391,32 @@ def get_delegated_hosts(self, result, task):
         host_name = result.get('_ansible_delegated_vars', {}).get('ansible_delegated_host', None)
         return [host_name or task.delegate_to]
 
+    def _set_always_delegated_facts(self, result, task):
```

Comment:
Minor nit, but the dual side-effect thing here feels kinda fragile in the face of future changes to this code (since it's both setting facts on delegated hosts and removing items from the original result). Maybe just return two dicts and handle the actual setting in the caller?

Reply:
I get what you mean, but after looking at it, I'd rather not go down this route.  The calling code, would have to be basically rewritten to support this.  I could make it potentially more clear if I just had it return `result`. But since the operation is mutable, it's unnecessary.